### PR TITLE
[CWS] Add support for mutable SECL variables

### PR DESCRIPTION
**Pull request by lebauce**
### What does this PR do?

Allow setting variables to store states through rule actions.

It is now possible to define actions for rules as follows:

```
- id: my_rule
  expression: ...
  actions:
    - set:
        name: my_boolean_variable
        value: true
    - set:
        name: my_string_variable
        value: true
    - set:
        name: my_other_variable
        field: process.file.name
```

These actions will be executed when the rule is triggered by an event.
Right now, only `set` actions can be defined.
`name` is the name of the variable that will be set by the actions.
The value for the variable can be specified
- using `value` for a predefined value
  (strings, integers, booleans, array of strings and array of integers are currently supported).
- using `field` to use the value of an event field.

Variable arrays can be modified by specifying `append: true`.

Variables can be reused in rule expressions like a regular variable:

```
- id: my_other_rule
  expression: |-
    open.file.path == ${my_other_variable}
```

By default, variables are global. They can be attached to a specific process by using the `process`
scope as follows:

```
- set:
    name: my_scoped_variable
    scope: process
    value: true
```

The variable can be referenced in other expressions as `${process.my_scoped_variable}`. When the process dies, the
variable with be automatically freed.

### Motivation

Variables can be used to store state, allowing features such as rule chaining
or process tagging.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/cws-secl-state-variables-8f37e28c65c30d03.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            value: true
+        - set:
+            name: my_other_variable
+            field: process.file.name
+    ```
+
+    These actions will be executed when the rule is triggered by an event.
+    Right now, only `set` actions can be defined.
+    `name` is the name of the variable that will be set by the actions.
+    The value for the variable can be specified
```
Comment (maycmlee) [inline releasenotes/notes/cws-secl-state-variables-8f37e28c65c30d03.yaml:26]:
```suggestion
    The value for the variable can be specified by using:
```

File: releasenotes/notes/cws-secl-state-variables-8f37e28c65c30d03.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    - using `value` for a predefined value
```
Comment (maycmlee) [inline releasenotes/notes/cws-secl-state-variables-8f37e28c65c30d03.yaml:27]:
```suggestion
    - `value` for a predefined value
```

File: releasenotes/notes/cws-secl-state-variables-8f37e28c65c30d03.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      (strings, integers, booleans, array of strings and array of integers are currently supported).
+    - using `field` to use the value of an event field.
```
Comment (maycmlee) [inline releasenotes/notes/cws-secl-state-variables-8f37e28c65c30d03.yaml:29]:
```suggestion
    -`field` for the value of an event field.
```

**Review by maycmlee — APPROVED** [review]
Couple of small changes.

**Review by djmitche — COMMENTED** [review]
It doesn't look like there's anything here for team/agent-core in the final analysis.  Let me know if I misunderstand!

File: pkg/security/secl/model/model.go [hunk 4]
```diff
@@ -279,6 +279,8 @@ type Process struct {
 	ArgsTruncated bool     `field:"args_truncated,ResolveProcessArgsTruncated"`                                                                            // Indicator of arguments truncation
 	Envs          []string `field:"envs,ResolveProcessEnvs:100"`                                                                                           // Environment variables of the process
 	EnvsTruncated bool     `field:"envs_truncated,ResolveProcessEnvsTruncated"`                                                                            // Indicator of environment variables truncation
+
+	Variables eval.Variables `field:"-"`
```
Comment (safchain) [inline pkg/security/secl/model/model.go:283]: is it still used ?

**Review by paulcacheux — APPROVED** [review]

**Pull request merged**